src/main/java/org/apache/accumulo/testing/continuous/ManySplits.java

```diff
@@ -0,0 +1,177 @@
+            NAMESPACE);
+      }
+
+      final NewTableConfiguration ntc = new NewTableConfiguration();
+      ntc.setProperties(Map.of(Property.TABLE_SPLIT_THRESHOLD.getKey(), splitThreshold));
+
+      log.info("Properties being used to create tables for this test: {}",
+          ntc.getProperties().toString());
+
+      String firstTable = tableNames.get(0);
+
+      log.info("Creating initial table: {}", firstTable);
+      try {
+        client.tableOperations().create(firstTable, ntc);
+      } catch (TableExistsException e) {
+        log.info("Test probably wont work if the table already exists with data present", e);
+      }
+
+      log.info("Ingesting {} entries into first table, {}.", initialData, firstTable);
+      ContinuousIngest.doIngest(client, rowMin, rowMax, firstTable, testProps, maxColF, maxColQ,
+          initialData, false, random);
+
+      client.tableOperations().flush(firstTable);
+
+      // clone tables instead of ingesting into each. it's a lot quicker
+      log.info("Creating {} tables by cloning the first", tableCount);
+      tableNames.stream().parallel().skip(1).forEach(tableName -> {
+        try {
+          client.tableOperations().clone(firstTable, tableName,
+              CloneConfiguration.builder().build());
+        } catch (TableExistsException e) {
+          log.info(
+              "table {} already exists. Continuing with existing table. This might mess with the expected values",
+              tableName);
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        }
+      });
+
+      Map<String,Integer> thresholdToExpectedTabletCount = new LinkedHashMap<>(3);
+      thresholdToExpectedTabletCount.put("100M", 4);
+      thresholdToExpectedTabletCount.put("10M", 32);
+      thresholdToExpectedTabletCount.put("1M", 512);
+
+      // main loop
+      // reduce the split threshold then wait for the expected number of tablets to appear
+      for (var entry : thresholdToExpectedTabletCount.entrySet()) {
+        String oldSplitThreshold = splitThreshold;
+        splitThreshold = entry.getKey();
+        final int expectedTabletCount = entry.getValue();
+
+        log.info("Changing split threshold on all tables from {} to {}", oldSplitThreshold,
+            splitThreshold);
+
+        long beforeThresholdUpdate = System.nanoTime();
+
+        // update the split threshold on all tables
+        tableNames.stream().parallel().forEach(tableName -> {
+          try {
+            client.tableOperations().setProperty(tableName, Property.TABLE_SPLIT_THRESHOLD.getKey(),
+                splitThreshold);
+          } catch (Exception e) {
+            throw new RuntimeException(e);
+          }
+        });
+
+        log.info("Waiting for tablet count on all tables to be greater than {}.",
+            expectedTabletCount - 1);
+
+        // wait for all tables to reach the expected number of tablets
+        tableNames.stream().parallel().forEach(tableName -> {
```

Comment:
This could instead scan the table computing the sum of file sizes for each tablet and comparing that to the tables configured split threshold.  Waiting for all tablets to be below the threshhold.  Use the file sizes form the tablet metadata because that is what the split code uses. If doing that then would not need to know an expected number of splits.

Reply:
Did this in b226faa